test/tests/librariesTest.js

```diff
@@ -0,0 +1,106 @@
+describe("Zotero.Libraries", function() {
+	it("should provide user library ID as #userLibraryID", function() {
+		assert.isDefined(Zotero.Libraries.userLibraryID, "#userLibraryID is defined");
+		assert(Number.isInteger(Zotero.Libraries.userLibraryID), "#userLibraryID is an integer");
+		assert.isAbove(Zotero.Libraries.userLibraryID, -1, "#userLibraryID is a positive integer");
+	});
+	it("should provide publications library ID as #publicationsLibraryID", function() {
+		assert.isDefined(Zotero.Libraries.publicationsLibraryID, "#publicationsLibraryID is defined");
+		assert(Number.isInteger(Zotero.Libraries.publicationsLibraryID), "#publicationsLibraryID is an integer");
+		assert.isAbove(Zotero.Libraries.publicationsLibraryID, -1, "#publicationsLibraryID is a positive integer");
+	});
+	it("should provide feed library ID as #feedLibraryID", function() {
+		assert.isDefined(Zotero.Libraries.feedLibraryID, "#feedLibraryID is defined");
+		assert(Number.isInteger(Zotero.Libraries.feedLibraryID), "#feedLibraryID is an integer");
+		assert.isAbove(Zotero.Libraries.feedLibraryID, -1, "#feedLibraryID is a positive integer");
+	});
+	describe("#getAll()", function() {
+		let ids = Zotero.Libraries.getAll();
+		
+		it("should return an array of valid library IDs", function() {
+			assert.isArray(ids, "returns an array");
+			assert(ids.reduce(function(res, id) { return res && Number.isInteger(id) && id >= 0 }, true), "All IDs are positive integers");
+		})
+		it("should return all library IDs", function *() {
```

Comment:
Interestingly, this currently fails https://travis-ci.org/zotero/zotero/jobs/65026201#L516 but I haven't investigated why yet.